alt_evaluator.py

```diff
@@ -12,17 +13,28 @@ def __init__(self, path_to_lmdb_dir):
             transforms.ToTensor(),
             transforms.Normalize([0.5, 0.5, 0.5], [0.5, 0.5, 0.5])
         ])
-        self._loader = torch.utils.data.DataLoader(Dataset(path_to_lmdb_dir, transform), batch_size=128, shuffle=False)
+        self.batch_size = 128
+        self._loader = torch.utils.data.DataLoader(Dataset(path_to_lmdb_dir, transform), batch_size=self.batch_size, shuffle=False)
 
-    def evaluate(self, model):
+    def evaluate(self, model, number_of_images_to_evaluate):
         results = []
 
+        number_of_batches_to_evaluate = math.inf
+        if number_of_images_to_evaluate:
+            number_of_batches_to_evaluate = math.ceil(int(number_of_images_to_evaluate) / self.batch_size)
+            print("Number of batches to evaluate: ", number_of_batches_to_evaluate)
+
         with torch.no_grad():
 
             for i, (images, length_labels, digits_labels, paths) in enumerate(self._loader):
                 images, length_labels, digits_labels = images.cpu(), length_labels.cpu(), [digit_labels.cpu() for digit_labels in digits_labels]
                 length_logits, digit1_logits, digit2_logits, digit3_logits, digit4_logits, digit5_logits = model.eval()(images)
-                print("Evaluating images in batch: ", i+1)
+
+                batch_num = i+1
+                if batch_num > number_of_batches_to_evaluate:
+                    break
```

Comment:
should this be higher so that it doesn't waste effort evaluating?

Reply:
number_of_batches_to_evaluated is determined by the number_of_images_to_evaluate:
```
number_of_batches_to_evaluate = math.inf
        if number_of_images_to_evaluate:
            number_of_batches_to_evaluate = math.ceil(int(number_of_images_to_evaluate) / self.batch_size)
            print("Number of batches to evaluate: ", number_of_batches_to_evaluate)
```

`number_of_images_to_evaluate` comes from the model academy api call:
```
params = {
        "dataset_path": base64.b64encode(b"dataset/data"),
    }
    if number_of_images_to_evaluate:
        params["number_of_images"] = number_of_images_to_evaluate
```

so the number of batches evaluated can be dynamically set by model academy, can be as many or as few as we like